test/Google.Datastore.V1Beta3.Tests/EntityTest.cs

```diff
@@ -77,5 +77,32 @@ public void Mutations_NoKey()
             Assert.Throws<InvalidOperationException>(() => entity.ToUpsert());
             Assert.Throws<InvalidOperationException>(() => entity.ToDelete());
         }
+
+        [Fact]
+        public void ArrayProperties()
+        {
+            // Demonstrating three ways of specifying arrays as entity properties
+            var variant1 = new Entity
+            {
+                ["tags"] = new[] { "a", "b", "c" }
```

Comment:
It's mostly the null handling of ArrayValue.GetEnumerator/Add, as discussed with Mike above. Removing that code would make variant2 invalid, but variant1 and variant3 would still be valid.